### PR TITLE
Fix C# code style

### PR DESCRIPTION
**Pull request by pepone**
This PR fixes the C# code style:

- Converted most of the files to use file scoped namespaces
- Reworked the checked-in generated files like Communicator.cs 
- Added dotnet format workflow to check the code style.

### CODE REVIEW

File: csharp/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,233 @@
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
+
+# C# files
+[*.cs]
+
+#### Guidelines ####
+guidelines = 120
+
+#### Core EditorConfig Options ####
```
Comment (bernardnormier) [inline csharp/.editorconfig:10]: I assume these are the same options as in icerpc-csharp?

File: csharp/src/Ice/Buffer.cs [hunk 2]
```diff
@@ -2,217 +2,213 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-namespace IceInternal
+namespace IceInternal;
+
+using System.Diagnostics;
```
Comment (bernardnormier) [inline csharp/src/Ice/Buffer.cs:7]: Should we move all these using at the top of the file? For a follow-up PR?

**Review by bernardnormier — APPROVED** [review]

File: csharp/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,233 @@
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
```
Comment (externl) [inline csharp/.editorconfig:2]: I think we should make a general top level one with this.

File: csharp/.editorconfig (hunk 1 above)
Comment (pepone) [reply csharp/.editorconfig:10]: Yes copied from icerpc-csharp repository.

**Review by externl — APPROVED** [review]
LGTM!

**Pull request merged**